exporter/management/commands/flattener.py

```diff
@@ -27,42 +27,52 @@ class Command(BaseCommand):
     """
 
     def handle(self, *args, **options):
-        consume(callback, "flattener_init", decorator=decorator)
+        consume(callback, "flattener_init", ["flattener_init", "flattener_file"], decorator=decorator)
 
 
 def callback(state, channel, method, properties, input_message):
     job_id = input_message.get("job_id")
-    max_rows_lower_bound = Job.objects.get(id=job_id).get_max_rows_lower_bound()
-
+    file_path = input_message.get("file_path")
     export = Export(job_id, export_type="flat")
-    export.lock()
 
     # Acknowledge now to avoid connection losses. The rest can run for hours and is irreversible anyhow.
     ack(state, channel, method.delivery_tag)
 
-    for entry in os.scandir(export.directory):
-        if not entry.name.endswith(".jsonl.gz") or "_" in entry.name:  # don't process months at the moment
-            continue
+    if file_path:
+        process_file(file_path, job_id)
+    else:
+        for entry in os.scandir(export.directory):
+            if not entry.name.endswith(".jsonl.gz") or "_" in entry.name:  # don't process months at the moment
+                continue
+            publish({"job_id": job_id, "file_path": entry.path}, "flattener_file")
+
+
+def process_file(file_path, job_id):
```

Comment:
Do we want to add any logic to skip a file if it exists? e.g. if we want to re-process Colombia if only a few files failed.

If we do add that logic, then we'll want to delete files in the `except` block in my other comment.

Reply:
Hmm, but if we delete files in the `except` block, then we'll never have a file to skip, or am I understanding it wrong?

Reply:
Ah, never mind, I forgot that we have a worker per file now, so each worker would delete the failed files only

Reply:
I've implemented this, some additional logic was needed you check if either the CSV or the Excel file exists already (or both). I anticipate that in the future when the Excel feature works correctly, we will need to re-run flattener for generating the excel files, so I've added the logic for not creating the CSV files again in these cases.

Reply:
I've upgraded flatterer and removed XLSX checks in https://github.com/open-contracting/data-registry/pull/255/commits/b7caf534018d2c984ab36cb7704960f1e14c0e20